### ISSUE TITLE
heap-buffer-overflow in cmsxform.c

### ISSUE THREAD

**Issue by treebacker** [mm2/Little-CMS#259]
In source code `cmsxform.c`, function **PrecalculatedXFORM**

Below code:

```c
    for (i = 0; i < LineCount; i++) {

        accum = (cmsUInt8Number*)in + strideIn;
        output = (cmsUInt8Number*)out + strideOut;

        for (j = 0; j < PixelsPerLine; j++) {

            accum = p->FromInput(p, wIn, accum, Stride->BytesPerPlaneIn);
            p->Lut->Eval16Fn(wIn, wOut, p->Lut->Data);
            output = p->ToOutput(p, wOut, output, Stride->BytesPerPlaneOut);
        }
```

`PixelsPerLine` May be larger than sizeof(output), which results a heap buffer overflow.

command line: ./utils/tificc/.libs/tificc bug1 /dev/null

 In AddressSanitizer:
![1](https://user-images.githubusercontent.com/44844446/115103569-5de9c400-9f85-11eb-9804-03348fd94c92.png)

In gdb:
![bug1](https://user-images.githubusercontent.com/44844446/115103572-65a96880-9f85-11eb-8906-5f5a28449b9b.png)

The crafted file is in the zip:
[bug1.zip](https://github.com/mm2/Little-CMS/files/6328694/bug1.zip)

**Comment by mm2** [comment]
Thanks for reporting.
However, what you are reporting may be a vulnerability of tifficc, which is a **sample** of how to use the little CMS library and **it is not required or used** in client applications using the library. So, I will try to add some checks as long as that would not affect code readability, which is the primary goal of this sample.

 **Please note any CVE regarding this kind of vulnerabilities would be promptly disputed, as are related with samples and not with the little cms component, which is unaffected**.

**Comment by mm2** [comment]
Fixed by 807809d7dd9b09c0380b71d5d5ebee3c1439ca99
